src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,87 @@
+  selectedTodo: Todo | null;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User>(defaultUser);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(setUser)
+      .catch(() => console.error('Error while trying to load user!'));
+  }, [selectedUserId]);
+
+  const isLoaded = selectedTodo && user !== defaultUser;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {!isLoaded ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{selectedTodo.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              onClick={() => dispatch(currentTodoSlice.actions.viewTodo(null))}
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {selectedTodo.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              <strong
+                className={cn({
+                  'has-text-success': selectedTodo.completed,
+                  'has-text-danger': !selectedTodo.completed,
+                })}
+              >
+                {selectedTodo.completed ? `Done` : `Planned`}
+              </strong>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
The optional chaining operator `?.` is used here, but since `user` is always set to a `User` object (either the default or fetched), the optional chaining is unnecessary. You can safely access `user.email` and `user.name` directly.